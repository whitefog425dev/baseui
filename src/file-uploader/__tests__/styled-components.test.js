/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

import React from 'react';
import {shallow} from 'enzyme';
import {
  StyledFileDragAndDrop,
  StyledContentMessage,
  StyledContentSeparator,
  StyledRoot,
  StyledFilesList,
  StyledHiddenInput,
  StyledAcceptedFile,
  StyledRejectedFile,
} from '../index.js';

describe('FileUploader styled components', () => {
  const styledComponents = [
    [StyledFileDragAndDrop, 'StyledFileDragAndDrop'],
    [StyledContentMessage, 'StyledContentMessage'],
    [StyledContentSeparator, 'StyledContentSeparator'],
    [StyledRoot, 'StyledRoot'],
    [StyledFilesList, 'StyledFilesList'],
    [StyledHiddenInput, 'StyledHiddenInput'],
    [StyledAcceptedFile, 'StyledAcceptedFile'],
    [StyledRejectedFile, 'StyledRejectedFile'],
  ];

  test.each(styledComponents)('default properties', (Component, name) => {
    const component = shallow(<Component />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has expected default styles`,
    );
  });

  test.each(styledComponents)('disabled', (Component, name) => {
    const component = shallow(<Component $isDisabled />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has expected styles when disabled`,
    );
  });

  test.each(styledComponents)('drag active', (Component, name) => {
    const component = shallow(<Component $isDragActive />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has expected styles when drag active`,
    );
  });

  test.each(styledComponents)('drag accept', (Component, name) => {
    const component = shallow(<Component $isDragAccept />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has expected styles when drag accept`,
    );
  });

  test.each(styledComponents)('drag reject', (Component, name) => {
    const component = shallow(<Component $isDragReject />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has expected styles when drag reject`,
    );
  });

  test.each(styledComponents)('focused', (Component, name) => {
    const component = shallow(<Component $isFocused />);
    expect(component.instance().getStyles()).toMatchSnapshot(
      `${name} has expected styles when focused`,
    );
  });
});
