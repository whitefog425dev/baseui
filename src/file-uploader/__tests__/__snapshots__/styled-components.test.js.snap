// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploader styled components default properties: StyledAcceptedFile has expected default styles 1`] = `Object {}`;

exports[`FileUploader styled components default properties: StyledContentMessage has expected default styles 1`] = `
Object {
  "fontFamily": "$theme.typography.font450.fontFamily",
  "fontSize": "$theme.typography.font450.fontSize",
  "fontWeight": "$theme.typography.font450.fontWeight",
  "lineHeight": "$theme.typography.font450.lineHeight",
}
`;

exports[`FileUploader styled components default properties: StyledContentSeparator has expected default styles 1`] = `
Object {
  "color": "$theme.colors.fileUploaderSeparatorColor",
}
`;

exports[`FileUploader styled components default properties: StyledFileDragAndDrop has expected default styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "$theme.colors.fileUploaderBackgroundColor",
  "borderColor": "$theme.colors.fileUploaderBorderColorDefault",
  "borderRadius": "$theme.borders.radius200",
  "borderStyle": "dashed",
  "borderWidth": "$theme.sizing.scale0",
  "boxSizing": "border-box",
  "display": "flex",
  "flexDirection": "column",
  "outline": null,
  "paddingBottom": "$theme.sizing.scale900",
  "paddingLeft": "$theme.sizing.scale800",
  "paddingRight": "$theme.sizing.scale800",
  "paddingTop": "$theme.sizing.scale900",
  "width": "100%",
}
`;

exports[`FileUploader styled components default properties: StyledFilesList has expected default styles 1`] = `Object {}`;

exports[`FileUploader styled components default properties: StyledHiddenInput has expected default styles 1`] = `Object {}`;

exports[`FileUploader styled components default properties: StyledRejectedFile has expected default styles 1`] = `
Object {
  "color": "$theme.colors.negative400",
}
`;

exports[`FileUploader styled components default properties: StyledRoot has expected default styles 1`] = `
Object {
  "fontFamily": "$theme.typography.font450.fontFamily",
  "fontSize": "$theme.typography.font450.fontSize",
  "fontWeight": "$theme.typography.font450.fontWeight",
  "lineHeight": "$theme.typography.font450.lineHeight",
}
`;

exports[`FileUploader styled components disabled: StyledAcceptedFile has expected styles when disabled 1`] = `Object {}`;

exports[`FileUploader styled components disabled: StyledContentMessage has expected styles when disabled 1`] = `
Object {
  "fontFamily": "$theme.typography.font450.fontFamily",
  "fontSize": "$theme.typography.font450.fontSize",
  "fontWeight": "$theme.typography.font450.fontWeight",
  "lineHeight": "$theme.typography.font450.lineHeight",
}
`;

exports[`FileUploader styled components disabled: StyledContentSeparator has expected styles when disabled 1`] = `
Object {
  "color": "$theme.colors.fileUploaderSeparatorColor",
}
`;

exports[`FileUploader styled components disabled: StyledFileDragAndDrop has expected styles when disabled 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "$theme.colors.fileUploaderBackgroundColor",
  "borderColor": "$theme.colors.fileUploaderBorderColorDefault",
  "borderRadius": "$theme.borders.radius200",
  "borderStyle": "dashed",
  "borderWidth": "$theme.sizing.scale0",
  "boxSizing": "border-box",
  "display": "flex",
  "flexDirection": "column",
  "outline": "none",
  "paddingBottom": "$theme.sizing.scale900",
  "paddingLeft": "$theme.sizing.scale800",
  "paddingRight": "$theme.sizing.scale800",
  "paddingTop": "$theme.sizing.scale900",
  "width": "100%",
}
`;

exports[`FileUploader styled components disabled: StyledFilesList has expected styles when disabled 1`] = `Object {}`;

exports[`FileUploader styled components disabled: StyledHiddenInput has expected styles when disabled 1`] = `Object {}`;

exports[`FileUploader styled components disabled: StyledRejectedFile has expected styles when disabled 1`] = `
Object {
  "color": "$theme.colors.negative400",
}
`;

exports[`FileUploader styled components disabled: StyledRoot has expected styles when disabled 1`] = `
Object {
  "fontFamily": "$theme.typography.font450.fontFamily",
  "fontSize": "$theme.typography.font450.fontSize",
  "fontWeight": "$theme.typography.font450.fontWeight",
  "lineHeight": "$theme.typography.font450.lineHeight",
}
`;

exports[`FileUploader styled components drag accept: StyledAcceptedFile has expected styles when drag accept 1`] = `Object {}`;

exports[`FileUploader styled components drag accept: StyledContentMessage has expected styles when drag accept 1`] = `
Object {
  "fontFamily": "$theme.typography.font450.fontFamily",
  "fontSize": "$theme.typography.font450.fontSize",
  "fontWeight": "$theme.typography.font450.fontWeight",
  "lineHeight": "$theme.typography.font450.lineHeight",
}
`;

exports[`FileUploader styled components drag accept: StyledContentSeparator has expected styles when drag accept 1`] = `
Object {
  "color": "$theme.colors.fileUploaderSeparatorColor",
}
`;

exports[`FileUploader styled components drag accept: StyledFileDragAndDrop has expected styles when drag accept 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "$theme.colors.fileUploaderBackgroundColor",
  "borderColor": "$theme.colors.fileUploaderBorderColorDefault",
  "borderRadius": "$theme.borders.radius200",
  "borderStyle": "dashed",
  "borderWidth": "$theme.sizing.scale0",
  "boxSizing": "border-box",
  "display": "flex",
  "flexDirection": "column",
  "outline": null,
  "paddingBottom": "$theme.sizing.scale900",
  "paddingLeft": "$theme.sizing.scale800",
  "paddingRight": "$theme.sizing.scale800",
  "paddingTop": "$theme.sizing.scale900",
  "width": "100%",
}
`;

exports[`FileUploader styled components drag accept: StyledFilesList has expected styles when drag accept 1`] = `Object {}`;

exports[`FileUploader styled components drag accept: StyledHiddenInput has expected styles when drag accept 1`] = `Object {}`;

exports[`FileUploader styled components drag accept: StyledRejectedFile has expected styles when drag accept 1`] = `
Object {
  "color": "$theme.colors.negative400",
}
`;

exports[`FileUploader styled components drag accept: StyledRoot has expected styles when drag accept 1`] = `
Object {
  "fontFamily": "$theme.typography.font450.fontFamily",
  "fontSize": "$theme.typography.font450.fontSize",
  "fontWeight": "$theme.typography.font450.fontWeight",
  "lineHeight": "$theme.typography.font450.lineHeight",
}
`;

exports[`FileUploader styled components drag active: StyledAcceptedFile has expected styles when drag active 1`] = `Object {}`;

exports[`FileUploader styled components drag active: StyledContentMessage has expected styles when drag active 1`] = `
Object {
  "fontFamily": "$theme.typography.font450.fontFamily",
  "fontSize": "$theme.typography.font450.fontSize",
  "fontWeight": "$theme.typography.font450.fontWeight",
  "lineHeight": "$theme.typography.font450.lineHeight",
}
`;

exports[`FileUploader styled components drag active: StyledContentSeparator has expected styles when drag active 1`] = `
Object {
  "color": "$theme.colors.fileUploaderSeparatorColor",
}
`;

exports[`FileUploader styled components drag active: StyledFileDragAndDrop has expected styles when drag active 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "$theme.colors.fileUploaderBackgroundColor",
  "borderColor": "$theme.colors.fileUploaderBorderColorActive",
  "borderRadius": "$theme.borders.radius200",
  "borderStyle": "dashed",
  "borderWidth": "$theme.sizing.scale0",
  "boxSizing": "border-box",
  "display": "flex",
  "flexDirection": "column",
  "outline": null,
  "paddingBottom": "$theme.sizing.scale900",
  "paddingLeft": "$theme.sizing.scale800",
  "paddingRight": "$theme.sizing.scale800",
  "paddingTop": "$theme.sizing.scale900",
  "width": "100%",
}
`;

exports[`FileUploader styled components drag active: StyledFilesList has expected styles when drag active 1`] = `Object {}`;

exports[`FileUploader styled components drag active: StyledHiddenInput has expected styles when drag active 1`] = `Object {}`;

exports[`FileUploader styled components drag active: StyledRejectedFile has expected styles when drag active 1`] = `
Object {
  "color": "$theme.colors.negative400",
}
`;

exports[`FileUploader styled components drag active: StyledRoot has expected styles when drag active 1`] = `
Object {
  "fontFamily": "$theme.typography.font450.fontFamily",
  "fontSize": "$theme.typography.font450.fontSize",
  "fontWeight": "$theme.typography.font450.fontWeight",
  "lineHeight": "$theme.typography.font450.lineHeight",
}
`;

exports[`FileUploader styled components drag reject: StyledAcceptedFile has expected styles when drag reject 1`] = `Object {}`;

exports[`FileUploader styled components drag reject: StyledContentMessage has expected styles when drag reject 1`] = `
Object {
  "fontFamily": "$theme.typography.font450.fontFamily",
  "fontSize": "$theme.typography.font450.fontSize",
  "fontWeight": "$theme.typography.font450.fontWeight",
  "lineHeight": "$theme.typography.font450.lineHeight",
}
`;

exports[`FileUploader styled components drag reject: StyledContentSeparator has expected styles when drag reject 1`] = `
Object {
  "color": "$theme.colors.fileUploaderSeparatorColor",
}
`;

exports[`FileUploader styled components drag reject: StyledFileDragAndDrop has expected styles when drag reject 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "$theme.colors.fileUploaderBackgroundColor",
  "borderColor": "$theme.colors.fileUploaderBorderColorDefault",
  "borderRadius": "$theme.borders.radius200",
  "borderStyle": "dashed",
  "borderWidth": "$theme.sizing.scale0",
  "boxSizing": "border-box",
  "display": "flex",
  "flexDirection": "column",
  "outline": null,
  "paddingBottom": "$theme.sizing.scale900",
  "paddingLeft": "$theme.sizing.scale800",
  "paddingRight": "$theme.sizing.scale800",
  "paddingTop": "$theme.sizing.scale900",
  "width": "100%",
}
`;

exports[`FileUploader styled components drag reject: StyledFilesList has expected styles when drag reject 1`] = `Object {}`;

exports[`FileUploader styled components drag reject: StyledHiddenInput has expected styles when drag reject 1`] = `Object {}`;

exports[`FileUploader styled components drag reject: StyledRejectedFile has expected styles when drag reject 1`] = `
Object {
  "color": "$theme.colors.negative400",
}
`;

exports[`FileUploader styled components drag reject: StyledRoot has expected styles when drag reject 1`] = `
Object {
  "fontFamily": "$theme.typography.font450.fontFamily",
  "fontSize": "$theme.typography.font450.fontSize",
  "fontWeight": "$theme.typography.font450.fontWeight",
  "lineHeight": "$theme.typography.font450.lineHeight",
}
`;

exports[`FileUploader styled components focused: StyledAcceptedFile has expected styles when focused 1`] = `Object {}`;

exports[`FileUploader styled components focused: StyledContentMessage has expected styles when focused 1`] = `
Object {
  "fontFamily": "$theme.typography.font450.fontFamily",
  "fontSize": "$theme.typography.font450.fontSize",
  "fontWeight": "$theme.typography.font450.fontWeight",
  "lineHeight": "$theme.typography.font450.lineHeight",
}
`;

exports[`FileUploader styled components focused: StyledContentSeparator has expected styles when focused 1`] = `
Object {
  "color": "$theme.colors.fileUploaderSeparatorColor",
}
`;

exports[`FileUploader styled components focused: StyledFileDragAndDrop has expected styles when focused 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "$theme.colors.fileUploaderBackgroundColor",
  "borderColor": "$theme.colors.fileUploaderBorderColorDefault",
  "borderRadius": "$theme.borders.radius200",
  "borderStyle": "dashed",
  "borderWidth": "$theme.sizing.scale0",
  "boxSizing": "border-box",
  "display": "flex",
  "flexDirection": "column",
  "outline": null,
  "paddingBottom": "$theme.sizing.scale900",
  "paddingLeft": "$theme.sizing.scale800",
  "paddingRight": "$theme.sizing.scale800",
  "paddingTop": "$theme.sizing.scale900",
  "width": "100%",
}
`;

exports[`FileUploader styled components focused: StyledFilesList has expected styles when focused 1`] = `Object {}`;

exports[`FileUploader styled components focused: StyledHiddenInput has expected styles when focused 1`] = `Object {}`;

exports[`FileUploader styled components focused: StyledRejectedFile has expected styles when focused 1`] = `
Object {
  "color": "$theme.colors.negative400",
}
`;

exports[`FileUploader styled components focused: StyledRoot has expected styles when focused 1`] = `
Object {
  "fontFamily": "$theme.typography.font450.fontFamily",
  "fontSize": "$theme.typography.font450.fontSize",
  "fontWeight": "$theme.typography.font450.fontWeight",
  "lineHeight": "$theme.typography.font450.lineHeight",
}
`;
